### PR TITLE
#54 Added support for the Gatekeeper External Data Provider

### LINKED ISSUE #54: Add support for the Gatekeeper External Data Provider
The [Gatekeeper External Data Provider](https://open-policy-agent.github.io/gatekeeper/website/docs/externaldata/#provider), like the other supported Webhook types, requires the `caBundle` for the server.

### PR DESCRIPTION
**Pull request by bdun1013**
Signed-off-by: bdunnigan <bdunnigan@clarityinnovates.com>

Fixes #54

### CODE REVIEW

**Comment by ritazh** [comment]
Thanks for the PR @bdun1013 
> --- FAIL: TestReconcileWebhook (3.72s)
    --- FAIL: TestReconcileWebhook/externaldataprovider (0.05s)
        rotator_test.go:222: 
            creating webhookConfig
            Unexpected error:
                <*meta.NoKindMatchError | 0xc000a9a080>: {
                    GroupKind: {
                        Group: "externaldata.gatekeeper.sh",
                        Kind: "Provider",
                    },
                    SearchedVersions: ["v1beta1"],
                }
                no matches for kind "Provider" in version "externaldata.gatekeeper.sh/v1beta1"
            occurred
FAIL

**Comment by bdun1013** [comment]
@ritazh @maxsmythe I updated dependencies and the tests are now passing

File: Makefile [hunk 1]
```diff
@@ -5,3 +5,8 @@ lint:
 
 test:
 	go test ./... -coverprofile cover.out
+
+crds:
+	mkdir -p "config/externaldata/crds"
+	curl -L -O --output-dir "config/externaldata/crds" "https://raw.githubusercontent.com/open-policy-agent/frameworks/master/constraint/config/crds/externaldata.gatekeeper.sh_providers.yaml"
+	curl -L -O --output-dir "config/externaldata/crds" "https://raw.githubusercontent.com/open-policy-agent/frameworks/master/constraint/config/crds/kustomization.yaml"
```
Comment (ritazh) [inline Makefile:12]: nit: newline

**Review by ritazh — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (bdun1013) [reply Makefile]: Added newline

**Comment by bdun1013** [comment]
@ritazh Are you able to merge and release 0.7?

File: pkg/rotator/rotator.go [hunk 2]
```diff
@@ -83,10 +85,11 @@ type WebhookInfo struct {
...
 	t2g := map[WebhookType]schema.GroupVersionKind{
-		Validating:    {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "ValidatingWebhookConfiguration"},
-		Mutating:      {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "MutatingWebhookConfiguration"},
-		CRDConversion: {Group: "apiextensions.k8s.io", Version: "v1", Kind: "CustomResourceDefinition"},
-		APIService:    {Group: "apiregistration.k8s.io", Version: "v1", Kind: "APIService"},
+		Validating:           {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "ValidatingWebhookConfiguration"},
+		Mutating:             {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "MutatingWebhookConfiguration"},
+		CRDConversion:        {Group: "apiextensions.k8s.io", Version: "v1", Kind: "CustomResourceDefinition"},
+		APIService:           {Group: "apiregistration.k8s.io", Version: "v1", Kind: "APIService"},
+		ExternalDataProvider: {Group: "externaldata.gatekeeper.sh", Version: "v1beta1", Kind: "Provider"},
```
Comment (maxsmythe) [inline pkg/rotator/rotator.go:92]: I wonder at what point it makes sense to replace the object-specific code with more generic code (e.g. watch this Kind, make sure certs live at path X)

**Review by maxsmythe — APPROVED** [review]
LGTM, one comment about how we may want to eventually make the watch/update loop generic if we keep adding kinds.

File: pkg/rotator/rotator.go (hunk 2 above)
Comment (bdun1013) [reply pkg/rotator/rotator.go:92]: That could make sense. @maxsmythe would you mind if the discussion/implementation is delegated to a separate issue?

Comment (maxsmythe) [reply pkg/rotator/rotator.go:92]: Not at all

**Comment by bdun1013** [comment]
@ritazh @maxsmythe I am unable to merge. Would one of you mind merging and releasing v0.7.0? Thanks!

**Pull request merged**